### PR TITLE
chore(react-datepicker-compat): Add vr-tests and bundle-size fixtures for DatePicker

### LINKED ISSUE #27207: DatePicker: Add vr tests
This issue keeps track of the addition of vr-tests for DatePicker/Calendar.

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR adds vr tests for DatePicker compat and bundle size fixtures.

## Related Issue(s)

- Fixes #27207

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/4f76849de438f2ee7842121ac3b8176ba1c0a2a5'>4f76849de438f2ee7842121ac3b8176ba1c0a2a5</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: apps/vr-tests-react-components/package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
     "@fluentui/react-card": "^9.0.4",
```
Comment (fabricteam) [inline apps/vr-tests-react-components/package.json:27]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 26 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=27351&commitId=e26f4e2e9de3ab5714d458d697a029e1e7f93891&env=prod)

✅ There was 26 screenshots added, 0 screenshots removed, 1903 screenshots unchanged, 0 screenshots with different dimensions and 0 screenshots with visible difference.


  <details><summary><b>unknown</b> 26 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|DatePicker Compat.DateRange- day - Dark Mode.chromium.png|0| Added|
|DatePicker Compat.DateRange- day - High Contrast.chromium.png|0| Added|
|DatePicker Compat.DateRange- day.chromium.png|0| Added|
|DatePicker Compat.DateRange- month - Dark Mode.chromium.png|0| Added|
|DatePicker Compat.DateRange- month - High Contrast.chromium.png|0| Added|
|DatePicker Compat.DateRange- month.chromium.png|0| Added|
|DatePicker Compat.DateRange- week - Dark Mode.chromium.png|0| Added|
|DatePicker Compat.DateRange- week - High Contrast.chromium.png|0| Added|
|DatePicker Compat.DateRange- week.chromium.png|0| Added|
|DatePicker Compat.DateRange- work week - Dark Mode.chromium.png|0| Added|
|DatePicker Compat.DateRange- work week - High Contrast.chromium.png|0| Added|
|DatePicker Compat.DateRange- work week.chromium.png|0| Added|
|DatePicker Compat.Required.chromium.png|0| Added|
|DatePicker Compat.Underlined and required.chromium.png|0| Added|
|DatePicker Compat.Underlined.chromium.png|0| Added|
|DatePicker Compat.With label.chromium.png|0| Added|
|DatePicker Compat.allowTextInput.chromium.png|0| Added|
|DatePicker Compat.default - Dark Mode.opened.chromium.png|0| Added|
|DatePicker Compat.default - Dark Mode.rest.chromium.png|0| Added|
|DatePicker Compat.default - High Contrast.opened.chromium.png|0| Added|
|DatePicker Compat.default - High Contrast.rest.chromium.png|0| Added|
|DatePicker Compat.default.opened.chromium.png|0| Added|
|DatePicker Compat.default.rest.chromium.png|0| Added|
|DatePicker Compat.marked dates - Dark Mode.chromium.png|0| Added|
|DatePicker Compat.marked dates - High Contrast.chromium.png|0| Added|
|DatePicker Compat.marked dates.chromium.png|0| Added|

</details>


<div id="vrtCommentfluentuiv9"/>

**Comment by khmakoto** [comment]
You should add `"bundle-size": "bundle-size measure"` as part of the `scripts` in `package.json` so the bundle size is correctly calculated and reported.

File: apps/vr-tests-react-components/src/stories/DatePickerCompat.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    'DateRange: work week',
+    () => <DatePicker calendar={{ dateRangeType: DateRangeType.Week }} popover={{ open: true }} />,
+    {
+      includeDarkMode: true,
+      includeHighContrast: true,
+    },
+  )
+  .addStory(
+    'DateRange: week',
+    () => <DatePicker calendar={{ dateRangeType: DateRangeType.WorkWeek }} popover={{ open: true }} />,
```
Comment (khmakoto) [inline apps/vr-tests-react-components/src/stories/DatePickerCompat.stories.tsx:45]: The names of these are swapped:

```suggestion
    'DateRange: week',
    () => <DatePicker calendar={{ dateRangeType: DateRangeType.Week }} popover={{ open: true }} />,
    {
      includeDarkMode: true,
      includeHighContrast: true,
    },
  )
  .addStory(
    'DateRange: work week',
    () => <DatePicker calendar={{ dateRangeType: DateRangeType.WorkWeek }} popover={{ open: true }} />,
```

File: apps/vr-tests-react-components/src/stories/DatePickerCompat.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { Steps, StoryWright } from 'storywright';
+import { DatePicker, DateRangeType } from '@fluentui/react-datepicker-compat';
+import { storiesOf } from '@storybook/react';
+import { TestWrapperDecoratorFixedWidth } from '../utilities/TestWrapperDecorator';
+
```
Comment (khmakoto) [inline apps/vr-tests-react-components/src/stories/DatePickerCompat.stories.tsx:6]: You should specify the `today` prop in all vr-tests that have an open `DatePicker`, if you do not they will probably change daily breaking the PR pipeline.

File: apps/vr-tests-react-components/src/stories/DatePickerCompat.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      steps={new Steps()
+        .snapshot('rest', { cropTo: '.testWrapper' })
+        .click('.datepicker-input')
+        .snapshot('opened', { cropTo: '.testWrapper' })
+        .end()}
+    >
+      {story()}
+    </StoryWright>
+  ))
+  .addStory('default', () => <DatePicker input={{ className: 'datepicker-input' }} />, {
```
Comment (behowell) [inline apps/vr-tests-react-components/src/stories/DatePickerCompat.stories.tsx:30]: Do you need `input={{ className: 'datepicker-input' }}` for all of the ones below, to make the `click('.datepicker-input')` step work?

Or if you don't want the steps to apply to all stories, you could have a separate `storiesOf` with the same name `'DatePicker Compat'`, but without the story steps. See [Radio.stories.tsx](https://github.com/microsoft/fluentui/blob/master/apps/vr-tests-react-components/src/stories/Radio.stories.tsx) for an example.

File: apps/vr-tests-react-components/src/stories/DatePickerCompat.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    },
+  )
+  .addStory(
+    'DateRange: month',
+    () => <DatePicker calendar={{ dateRangeType: DateRangeType.Month }} popover={{ open: true }} />,
+    {
+      includeDarkMode: true,
+      includeHighContrast: true,
+    },
+  )
```
Comment (behowell) [inline apps/vr-tests-react-components/src/stories/DatePickerCompat.stories.tsx:65]: nit: You could probably get by with including DarkMode/HighContrast on just one of these DateRange stories.

**Review by behowell — APPROVED** [review]

**Pull request merged**